proposed/http-message.md

```diff
@@ -78,13 +78,13 @@ referring to these interfaces.
 
 #### 1.2 HTTP Headers
 
-##### Case-insensitive header names
+##### Case-insensitive header field names
 
-HTTP messages include case-insensitive header names. Headers are retrieved by name from
-classes implementing the `MessageInterface` interface in a case-insensitive
-manner. For example, retrieving the "foo" header will return the same result as
-retrieving the "FoO" header. Similarly, setting the "Foo" header will overwrite
-any previously set "foo" header.
+HTTP messages include case-insensitive header field names. Headers are retrieved by
+name from classes implementing the `MessageInterface` interface in a case-insensitive
```

Comment:
Remove `interface` following the interface name; the name contains that verbiage already.